# Data Analytics Task: E-commerce Sales Analysis
### Scenario:
You are a data analyst at an e-commerce company. Your task is to analyze the past month's sales data across various product categories. The dataset includes information on the ProductID, Category, UnitsSold, and UnitPrice for each sale.

### Objective:
- Calculate the total revenue for each product category.
- Determine the best-selling product in each category.
- Find the overall top-selling product based on units sold.
- Identify the category generating the highest revenue.
- Analyze sales trends and provide insights for potential inventory adjustments.

In [7]:
import pandas as pd

data = pd.read_csv("./modules/ecommerce_sales.csv")
data = data.dropna()
data

ProductID        Category  UnitsSold  UnitPrice
0      P001     Electronics       20.0     299.99
1      P002  Home & Kitchen       35.0      89.99
2      P003     Electronics       15.0     199.99
3      P004           Books       60.0       9.99
4      P005     Electronics       30.0     149.99
5      P006           Books       40.0      19.99
6      P007  Home & Kitchen       25.0      79.99
7      P008           Books       50.0      14.99
8      P009     Electronics       10.0     399.99
9      P010  Home & Kitchen       20.0      99.99

In [14]:
# Sum of all categories
data['Total Revenue'] = data["UnitsSold"] * data["UnitPrice"]

# Get only unique categories
category_revenue = data.groupby('Category')['Total Revenue'].sum()
category_revenue

Category
Books              2148.50
Electronics       17499.25
Home & Kitchen     7149.20
Name: Total Revenue, dtype: float64

In [17]:
# idxmax to get the highest's index
best_selling_products = data.loc[data.groupby("Category")['UnitsSold'].idxmax()]
best_selling_products

ProductID        Category  UnitsSold  UnitPrice  Total Revenue
3      P004           Books       60.0       9.99         599.40
4      P005     Electronics       30.0     149.99        4499.70
1      P002  Home & Kitchen       35.0      89.99        3149.65

In [21]:
top_selling_products = data.loc[data['UnitsSold'].idxmax()]
top_selling_products

ProductID         P004
Category         Books
UnitsSold         60.0
UnitPrice         9.99
Total Revenue    599.4
Name: 3, dtype: object

In [23]:
category_revenue.idxmax()

'Electronics'

### **Interpretation of Results**

**Total Revenue By Category:**

- **Books**: 2148.50
- **Electronics**: 17499.25
- **Home & Kitchen**: 7149.20

The total revenue per category shows that Electronics sold the most compared to the rest.
It suggests that Electronics are the highest in demand, price, or both compared to the other appliances.

**Overall Top-Selling Product:**

- Product ID P004, selling 60 units. While electronics generate more revenue, the highest sales were observed for P004 of the Books category.

**Best-Selling Product In Each Category:**

- **Books**: Product ID P004 with 60 units sold, generating $599.40 in revenue.
- **Electronics**: Product ID P005 with 30 units sold, generating $4,499.70 in revenue.
- **Home & Kitchen**: Product ID P002 with 35 units sold, generating $3,149.65 in revenue.

### **Recommendations**

- **Focus on High-Value categories**: Given that electronics makes the most revenue, consider increasing inventory for popular electronic appliances.
- **Volume vs Value**: Books show potential in terms of volume. Strategies such as bundling books with related electronics or home and kitchen products with cookbooks could increase sales across categories.
- **Promotional Activities**: For the Home and Kitchen category, which has a moderate performance in both revenue and units sold, target promitions or discounts could stimulate additional sales.
- **Customer Feedback**: Gather customer feedback, specifically for top-selling and high-revenue products to understand what drives their success. Use these insights to improve or highlight similar products in marketing campaigns
- **Inventory Management**: Adjust inventory levels based on sales data to ensure that high-demand products are always in stock, especially in the Electronics category, to maximize revenue potential.